pgvectorscale/src/access_method/pg_vector.rs

```diff
@@ -156,3 +205,65 @@ impl PgVector {
         unsafe { (*self.full_distance.unwrap()).to_slice() }
     }
 }
+
+impl Clone for PgVector {
+    fn clone(&self) -> Self {
+        unsafe {
+            let index_distance = self
+                .index_distance
+                .map(|original| Self::clone_inner(original));
+
+            let full_distance = if let Some(original) = self.full_distance {
+                // Check if full_distance points to the same memory as index_distance
+                if self.index_distance.is_some()
+                    && std::ptr::eq(original, self.index_distance.unwrap())
+                {
+                    // Reuse the same cloned pointer
+                    index_distance
+                } else {
+                    // Clone separately
+                    Some(Self::clone_inner(original))
+                }
+            } else {
+                None
+            };
+
+            PgVector {
+                index_distance,
+                index_distance_needs_pfree: index_distance.is_some(),
+                full_distance,
+                full_distance_needs_pfree: full_distance.is_some()
+                    && !std::ptr::eq(
+                        full_distance.unwrap_or(std::ptr::null_mut()),
+                        index_distance.unwrap_or(std::ptr::null_mut()),
+                    ),
+            }
+        }
+    }
+}
+
+impl PgVector {
+    unsafe fn clone_inner(original: *mut PgVectorInternal) -> *mut PgVectorInternal {
+        let dim = (*original).dim;
+        let slice = (*original).to_slice();
+
+        // Calculate total size needed: header + array of f32s
+        let header_size = std::mem::size_of::<PgVectorInternal>();
+        let array_size = dim as usize * std::mem::size_of::<f32>();
+        let total_size = header_size + array_size;
+
+        // Allocate new PostgreSQL memory
+        let new_ptr = pg_sys::palloc(total_size) as *mut PgVectorInternal;
+
+        // Copy the header
+        (*new_ptr).vl_len_ = (*original).vl_len_;
+        (*new_ptr).dim = dim;
+        (*new_ptr).unused = (*original).unused;
```

Comment:
It is undefined behavior to read uninitialized memory, including padding bytes. I think the fix is to make `unused` a `MaybeUninit<i16>` instead of an `i16` here:
https://github.com/timescale/pgvectorscale/blob/df48032af6ae348f8e79be7b40be7fb0ccc7a8b5/pgvectorscale/src/access_method/pg_vector.rs#L13